lib/collapsers/binary.js

```diff
@@ -1,15 +1,13 @@
 'use strict';
-var logger = require('bole')('collapsify:collapsers:binary');
-var base64Utils = require('../utils/base-64');
+const dataURI = require('../utils/data-uri');
 
-function external(url, opts) {
-  logger.info('Fetching external binary from "%s".', url);
-
-  return opts.fetch(url).then(collapse);
+async function external({fetch, resourceLocation: url}) {
+  const {contentType, body} = await fetch(url);
```

Comment:
Servers don't always return content-types browsers accept for dataURIs. It might be better to prefer a list for common file extensions, and fall back to the content-type if unknown.

Reply:
No, I definitely don't want to "guess" content-type again, it's a very insecure practice, and there is a reason browsers don't do that. We shouldn't change how it treats the content just by server-side extensions - if real browser wouldn't show the image on original HTML because of incorrect Content-Type, it shouldn't do that on the resulting HTML either.

> Servers don't always return content-types browsers accept for dataURIs. 

Can you provide an example of content-type that would work on a network response but not dataURI (after removing spaces)? From what I've seen, anything should work; moreover, as per spec, it's optional and even data URIs without any content-type like `data:,123` are perfectly valid.

Reply:
I'm definitely not saying we should guess, that was a mistake that was going to be the first thing I removed if I ever touched this again.

I'll have to dig through my archives to see if I can find the cases I found before.